worker/client.go

```diff
@@ -208,23 +212,30 @@ func (c *Client) object(ctx context.Context, path string, w io.Writer, entries *
 }
 
 // ObjectEntries returns the entries at the given path, which must end in /.
-func (c *Client) ObjectEntries(ctx context.Context, path string) (entries []api.ObjectMetadata, err error) {
-	err = c.object(ctx, path, nil, &entries)
+func (c *Client) ObjectEntries(ctx context.Context, path, prefix string, offset, limit int) (entries []api.ObjectMetadata, err error) {
+	path = strings.TrimPrefix(path, "/")
+	err = c.object(ctx, path, prefix, offset, limit, nil, &entries)
 	return
 }
 
 // DownloadObject downloads the object at the given path, writing its data to
 // w.
 func (c *Client) DownloadObject(ctx context.Context, w io.Writer, path string) (err error) {
-	path = strings.TrimLeft(path, "/")
-	err = c.object(ctx, path, w, nil)
+	if strings.HasSuffix(path, "/") {
+		return errors.New("the given path is a directory, use ObjectEntries instead")
```

Comment:
Shouldn't this be a check on the backend? Or do you want the API to still serve allow this?
Also why prevent directories in `DownloadObject` but not files in `ObjectEntries`?

Reply:
`DownloadObject` and `ObjectEntries` are syntactic sugar in the client, both call to `worker/objects` so a check on the backend is not possible but also not necessary. I just wanted to inform the user that he's using the client wrong. I added a similar error to `ObjectEntries`.

We can also just remove both but I would keep them personally. I think it's nice if a client picks up on me using it wrong and then directs me to whatever method I need to use.